### PR TITLE
Introducing measureDistance(to:xOffset:yOffset:) and the cooresponding |--| operator

### PR DESCRIPTION
**Pull request by dfed**
Addresses #3 

Ready for comment/review!

cc @shawnwelch @nickentin @abrons

### CODE REVIEW

File: README.md
Comment (NickEntin) [inline README.md:131]: Why does `.right` refer to `a`'s superview?

File: Tests/SubviewDistributionTests.swift
Comment (NickEntin) [inline Tests/SubviewDistributionTests.swift:49]: Would it be better to leave the PixelRounder tests using the old way of testing, and then add tests specifically for the new `|--|` operator?  That way each test would only be testing one area of functionality.

File: README.md
Comment (dfed) [reply README.md:131]: an unnamed `.<position>` always refers to the superview of the guy on the other side of the operator. It's a handy little shortcut. Works here, and also for alignment. e.g. `a.right --> .right`

If you think it's more confusing than helpful to have that in the README, I can change the example. Could also add an explanation for that shortcut to the README 🙂

File: Tests/SubviewDistributionTests.swift
Comment (dfed) [reply Tests/SubviewDistributionTests.swift:49]: It would be better. But I do like having this test be as readable as it is. Also, since alignment doesn't depend on the measuring code, I think we're good to go here.

File: README.md
Comment (NickEntin) [reply README.md:131]: Actually I really like that shortcut, but I would definitely make it clearer in the README.

File: Sources/Operators.swift
Comment (NickEntin) [inline Sources/Operators.swift]: What about `public func -(lhs: UIOffset, rhs: CGSize) -> CGSize`?

Comment (dfed) [reply Sources/Operators.swift]: Currently I've provided the ability to inset a `CGSize` by a `UIOffset` via the `-(lhs: CGSize, rhs: UIOffset) -> CGSize` operator.

If I allow you to `-(lhs: UIOffset, rhs: CGSize) -> CGSize`, then I'm allowing you to create a `CGSize` by insetting a `UIOffset` with a `CGSize`; that feels like a recipe for unexpected/accidental behavior.

Comment (NickEntin) [reply Sources/Operators.swift]: I agree it's probably not the common behavior (although I can imagine a layout where you may want to use it), but either way I think it's unexpected (missing) behavior to provide `CGSize + UIOffset`, `UIOffset + CGSize`, and `CGSize - UIOffset`, but not `UIOffset - CGSize`.

Comment (dfed) [reply Sources/Operators.swift]: I'm leaning towards leaving it as-is for now to avoid confusion – it's always easy to add features, but much harder to remove them.

I'm also tempted to remove `-` altogether and just make people use `+ -5.horizontalOffset` to make the math explicit. Thoughts?

Comment (NickEntin) [reply Sources/Operators.swift]: I like it.  Wouldn't be surprised if people complain, but I think it adds a bit of clarity that'll help in debugging.

File: README.md
Comment (dfed) [reply README.md:131]: This is actually in the README. See the third example under [Programatic Layout : Positioning Views : Examples](https://github.com/dfed/Relativity/blob/1b50d0ed53793af42708c9858f61d79d7baab3e7/README.md#examples).

If we want better examples / more documentation here I'll add that in a followup PR.

**Comment by dfed** [comment]
I'll want to adopt the learnings from #8 before I merge this.

**Comment by dfed** [comment]
Learnings from #8 have been adopted! Could use an LGTM before merge.

File: README.md
Comment (kathawthorne) [reply README.md:131]: I agree that it's a handy shortcut. I think the example here is fine because you clearly state in the comment above that you're comparing it to the superview, but maybe to make it more explicit you could use the superview's left side or something distinct so that when reading the comment and the code it's clear what goes with what.

File: Tests/SubviewDistributionTests.swift
Comment (kathawthorne) [reply Tests/SubviewDistributionTests.swift:49]: I agree that there should be explicit tests for `|--|` rather than just changing them here, but this is enough of a sanity check that it works as expected.

**Review by kathawthorne — APPROVED** [review]

File: Tests/SubviewDistributionTests.swift
Comment (dfed) [reply Tests/SubviewDistributionTests.swift:49]: After merging #8 I made the underlying measuring code rely on the same logic that `-->` uses. Writing some tests. Will push them before I merge.

**Pull request merged**
